cli/parser/parser.go

```diff
@@ -32,6 +32,25 @@ const (
 )
 
 var (
+	cliCommands = map[string]struct{}{
+		"add":      {},
+		"analyze":  {},
+		"ask":      {},
+		"wtf":      {},
+		"huh":      {},
+		"download": {},
+		"execute":  {},
+		"fix":      {},
+		"install":  {},
+		"login":    {},
+		"logout":   {},
+		"print":    {},
+		"remote":   {},
+		"update":   {},
+		"upload":   {},
+		"version":  {},
+	}
```

Comment:
There are starting to be quite a few places in the code that we have to update when adding a new command -

- The switch case in main.go
- This map
- The "help" map in `help.go`

We've forgotten to update the `help` section a few times before and it seems somewhat likely we can forget to update this too.

wdyt about adding a centralized command registry like

```
var Commands = []Command{
  {
    Name: "ask",
    Help: "...",
    Handler: ask.HandleAsk, // func(args []string) (exitCode int, err error)
    Aliases: []string{"wtf", "huh"},
  },
  // ...
}
```

then this `cliCommands` map, `help` output etc. can be derived from that registry.

Reply:
Good suggestion - refactored

The one slightly messy part is that we have to handle the help command separately. Otherwise there will be a circular dependency because the commands package will import help for the help handler. And the help handler imports commands because it needs the full list of commands